src/lib.rs

```diff
@@ -1329,6 +1367,31 @@ impl<'ctx> Bindings<'ctx> {
         try!(eof(&mut ps.s));
         ps.s.out.flush()
     }
+
+    /// Checks if format_bindings is set and runs rustfmt on the file
+    fn format_generated_file(&self, file: &Path) {
+        if !self.context.options().format_bindings {
+            return;
+        }
+
+        let rustfmt = if let Ok(rustfmt) = which::which("rustfmt") {
+            rustfmt
+        } else {
+            error!("Could not find rustfmt in the global path.");
+            return;
+        };
+
+        let mut cmd = Command::new(rustfmt);
+
+        if let Some(path) = self.context.options().format_configuration_file.as_ref().and_then(
+            |f| f.to_str()) {
+            cmd.args(&["--config-path", path]);
+        }
+
+        if let Err(e) = cmd.arg(file).status() {
```

Comment:
As far as I know, rustfmt does not write only the half. The problem with the status code is that if rustfmt can not format (because line too long or whatever), it ends with status != 0. I also can not return an error here, because write_to_file returns a io::Result. Chain_error would be nice here.